lib/AST/RequirementMachine/RequirementMachineRequests.cpp

```diff
@@ -667,7 +681,20 @@ InferredGenericSignatureRequestRQM::evaluate(
 
   if (ctx.LangOpts.RequirementMachineInferredSignatures ==
       RequirementMachineMode::Enabled) {
-    hadError |= diagnoseRequirementErrors(ctx, errors, allowConcreteGenericParams);
+    // Diagnose invalid requirements dropped during desugaring.
+    hadError |= diagnoseRequirementErrors(ctx, errors,
```

Comment:
This would be a bit cleaner if all errors were collected into a single vector owned by the request.